runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaDoFnRunners.java

```diff
@@ -439,6 +445,25 @@ public void finishBundle() {
       }
     }
 
+    private void closeBundle() throws Exception {
+      long bundleProcessingTimeout = pipelineOptions.getBundleProcessingTimeout();
+      if (bundleProcessingTimeout < 0) {
+        // RemoteBundle close blocks until all results are received
+        remoteBundle.close();
+      } else {
+        CompletableFuture<Void> future =
+            CompletableFuture.runAsync(
+                () -> {
+                  try {
+                    remoteBundle.close();
+                  } catch (Exception e) {
+                    throw new RuntimeException(e);
+                  }
+                });
+        future.get(bundleProcessingTimeout, TimeUnit.MILLISECONDS);
```

Comment:
This would throw `TimeoutException`/`ExecutionException` and mask the original cause in case of latter.
I'd preferably bubble up the original cause so that we know what is wrong in case of failure.